lib/index.js

```diff
@@ -95,15 +115,85 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
       res.status(204).end();
     }));
 
+  app.post(
+    routes.authenticate,
+    //validate('services.authn.token.postAuthenticate'),
+    asyncHandler(async (req, res) => {
+      const {type, account, email, hash, challenge} = req.body;
+
+      // get previously authenticated methods from session
+      let data = _getSessionAuthenticationData({req});
+      const authenticatedMethods = data.authenticatedMethods || [];
+
+      // make sure session data includes whether or not token client has
+      // registered
+      if(!authenticatedMethods.includes('token-client-registration')) {
+        // FIXME: get client ID from cookie
+        const clientId = 'FIXME';
+        const result = await brAuthnToken.clients.isRegistered(
+          {actor: null, email, account, clientId});
+        if(result.registered) {
+          authenticatedMethods.push('token-client-registration');
+        }
+      }
+
+      const result = await brAuthnToken.verify({
+        actor: null, account, email, type, hash, challenge,
+        authenticatedMethods
+      });
+      if(!result) {
+        // authentication failed
+        throw new BedrockError(
+          'Not authenticated.',
+          'NotAllowedError', {public: true, httpStatusCode: 400});
+      }
+
+      // update authenticated methods
+      const {authenticationMethod} = result.token;
+      if(!authenticatedMethods.includes(authenticationMethod)) {
+        authenticatedMethods.push(authenticationMethod);
+      }
+
+      // if `authenticationMethod` is `token-client-registration` then
+      // register token client for it
+      if(authenticationMethod === 'token-client-registration') {
+        // FIXME: get client ID from cookie
+        const clientId = 'FIXME';
```

Comment:
What's the situation here, should there be an issue?

Reply:
Yes, there should be an issue to implement an actual cookie for that.